### ISSUE TITLE
Missing windowsvista style plugin for Windows

### ISSUE THREAD

**Issue by jarolrod** [bitcoin/bitcoin#22132]
Here's a comparison of the Bitcoin Core GUI cross-compiled from linux to Windows between `master` and `0.21`:

| Master     | 0.21         |
| ------------- | ------------- |
| ![master-cross](https://user-images.githubusercontent.com/23396902/120451507-81af7080-c35f-11eb-825b-3bab15409c0b.png) | ![021-cross](https://user-images.githubusercontent.com/23396902/120453344-f3d48500-c360-11eb-8b97-c240f2ae7b7f.png) |

Yep, there's something wrong here. We can gain some insight into what's going on with the style by leveraging the logging of static plugins introduced in 957895c715f86feaa26c806e5fa8ebb77430a926. The `master` branch already includes this commit. I cherry-picked it on top of the `0.21` branch [here](https://github.com/jarolrod/bitcoin/tree/021-static). Now if we take a look at the `debug.log` for each, we'll see the following

*0.21 Branch:*
```
2021-06-02T07:23:59Z Qt 5.9.8 (static), plugin=windows (static)
2021-06-02T07:23:59Z Static plugins:
2021-06-02T07:23:59Z  QWindowsIntegrationPlugin, version 329992
2021-06-02T07:23:59Z Style: windowsvista / QWindowsVistaStyle
```

*Master Branch:*
```
2021-06-02T04:16:06Z Qt 5.12.10 (static), plugin=windows (static)
2021-06-02T04:16:06Z Static plugins:
2021-06-02T04:16:06Z  QWindowsIntegrationPlugin, version 330752
2021-06-02T04:16:06Z Style: windows / QWindowsStyle
```

We notice that the `master` branch's style is set to `windows` while the `0.21` branch is set to `windowsvista` style. Looking at the [Qt Widgets Gallery](https://doc.qt.io/qt-5/gallery.html), you can observe that `windows` style looks like Windows 95 while `windowsvista` is the "sleeker"/"modern" [QStyle](https://doc.qt.io/qt-5/qstyle.html) for the Windows platform.

This is what has changed between `master` and `0.21` and what is causing the GUI to look straight out of the 90's (maybe this is a retro release? 🕺). 

Now, what is causing this change? Well, I'm not completely sure. Otherwise I'd open up a PR. I do have some findings:

#### Different available QStyle Keys between Qt 5.9.8 and Qt 5.12 at runtime:

Looking at the `Qt Widgets` section of the depends Qt configuration output, we can see the following styles are available:

*0.21 Branch:*
```
Qt Widgets:
  GTK+ ................................... no
  Styles ................................. Fusion Windows WindowsXP WindowsVista
```

*Master Branch:*
```
Qt Widgets:
  GTK+ ................................... no
  Styles ................................. Fusion Windows WindowsVista
```

So while it seems that in regards to how Qt is configured, `WindowsVista` style is available in both cases,  If we apply the patch below to see the available [QStyle Keys](https://doc.qt.io/qt-5/qstylefactory.html#keys) at runtime, we don't see `windowsvista` as an available key on `master`: 
```
qInfo() << QStyleFactory::keys().join(",");
```

*0.21 Branch:*
```
2021-06-02T17:43:35Z GUI: "Windows,WindowsXP,WindowsVista,Fusion"
```

*Master Branch:*
```
2021-06-02T17:03:11Z GUI: "Windows,Fusion"
```

This means that we can't use the line below to fix the issue, and so a fix would involve changes to our build system: 
```
QApplication::setStyle("windowsvista");
```

**Windows Vista Style moved to a Plugin in Qt 5.10:**

This is based off what I found while looking for this issue online:

[Source A](https://trac.videolan.org/vlc/ticket/20780)

> I have also ran into this issue with Qt 5.11.1. Qt's Windows Vista style has been moved out to a separate DLL, which is styles\qwindowsvistastyle.dll. If you add this to the package, standard windows will get back their modern look. 

[Source B](https://stackoverflow.com/questions/49587418/qt-application-becomes-windows-old-styled-after-build-on-a-buildbot)

> I found a problem. After Qt5.10.1 the "windowsvista" style becomes a plugin. So in the final deploy file windowsvistastyle.dll schould exists. I add a folder "/styles" with this .dll and a problem is gone.

[Source C](https://github.com/Ruin0x11/tiled/commit/7537fd06187591ec4d2feecbdb1fe8da88ccf5a7)

> Ship style plugins, since native styles were moved into plugins as of Qt 5.10. This includes Windows Vista style on Windows and Adwaita on Linux.

If we figure out what needs to be done on the build side to link to this plugin, then we can use a similar patch as [vlc](http://git.videolan.org/?p=vlc.git;a=blobdiff;f=modules/gui/qt/qt.cpp;h=6fdbda06fa8f8aba4c9d43eef29543b28fdeedea;hp=1128633386a3fa43fe9a59ea5f633557a9596483;hb=0630bfa0436105ae7e24051cc16031c8ce2d77f5;hpb=07c35dcde98e25fba642085ea89080d94a0698d8) to fix this and so the following is a solution:

```
 Q_IMPORT_PLUGIN(QWindowsVistaStylePlugin)
```

**Comment by jarolrod** [comment]
cc @hebasto @fanquake

**Comment by prayank23** [comment]
> you can observe that windows style looks like Windows 95

I had noticed this few days back while testing something. It was intermittent, don't even remember if it was master or 0.21 or other branch and I ignored it assuming something wrong with my system itself (Windows can behave weird at times and most of the issues are fixed with restart)

Thanks for doing the research. Hope it is resolved soon.

**Comment by hebasto** [comment]
Fixed in #22133.
